### PR TITLE
Updated related components

### PR DESCRIPTION
**Pull request by andgen404**
### :pushpin: Summary

If merged, this PR will update related links in the `related` area for multiple components and patterns.

### :hammer_and_wrench: Detailed description

This Pull Request will correct any missing, unrelated, or inaccurately linked components and patterns by updating them with the appropriate connections.

### :link: External links

Jira ticket: [HDS-2393](https://hashicorp.atlassian.net/browse/HDS-2393)
Figma file: [FigJam Audit file](https://www.figma.com/file/bLbqBFEoXEnBPP5ZfuifpB/Audit-related-component-data-on-the-website?type=whiteboard&node-id=0%3A1&t=63GCgJJxlYA9gUU8-1)

***

### 👀 Reviewer's checklist:

- [x] +1 Percy if applicable
- [x] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed
- [x] Confirm that A11y tests have been run locally for this component

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2393]: https://hashicorp.atlassian.net/browse/HDS-2393?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by heatherlarsen — APPROVED** [review]
Double-checked it against the audit and it looks good 👍🏻

File: website/docs/components/flyout/index.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ caption: Displays additional details and information about an item or object, ov
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=23645%3A53756
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/flyout
+related: ['components/reveal','components/modal','components/accordian']
```
Comment (alex-ju) [inline website/docs/components/flyout/index.md:8]:
```suggestion
related: ['components/reveal','components/modal','components/accordion']
```

File: website/docs/components/form/checkbox/index.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ caption: A form element that allows users to select one or more items from a gro
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=9120%3A23132&t=pDgL7LJUJXZUN7Xq-3
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/form/checkbox
+related: ['components/form/toggle','components/form/radio',components/form/select']
```
Comment (alex-ju) [inline website/docs/components/form/checkbox/index.md:8]:
```suggestion
related: ['components/form/toggle','components/form/radio','components/form/select']
```

File: website/docs/components/form/radio/index.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ caption: A form element that allows users to select a single item from group of
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=13181%3A36977&t=pDgL7LJUJXZUN7Xq-3
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/form/radio
-related: ['components/form/checkbox', 'components/form/toggle', 'patterns/form-patterns']
+related: ['components/form/checkbox', 'components/form/toggle', 'patterns/form/select']
```
Comment (alex-ju) [inline website/docs/components/form/radio/index.md:8]:
```suggestion
related: ['components/form/checkbox', 'components/form/toggle', 'components/form/select']
```

File: website/docs/components/modal/index.md [hunk 4]
```diff
@@ -5,6 +5,7 @@ caption: A pop-up window used to request information, confirm a decision, or pro
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=22928%3A56204&t=pDgL7LJUJXZUN7Xq-1
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/modal
+related: ['components/alert','components/flyout','components/accordian','components/tooltip']
```
Comment (alex-ju) [inline website/docs/components/modal/index.md:8]:
```suggestion
related: ['components/alert','components/flyout','components/accordion','components/tooltip']
```

File: website/docs/components/reveal/index.md [hunk 5]
```diff
@@ -5,7 +5,7 @@ caption: A toggle that reveals additional content to the user when triggered.
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?type=design&node-id=35103%3A70091&t=PdXIqxQkqpsXUQki-1
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/reveal
-related: ['components/flyout']
+related: ['components/flyout','components/accordian','components/tooltip']
```
Comment (alex-ju) [inline website/docs/components/reveal/index.md:8]:
```suggestion
related: ['components/flyout','components/accordion','components/tooltip']
```

File: website/docs/components/tabs/index.md [hunk 6]
```diff
@@ -5,6 +5,7 @@ caption: Allows users to move among different views within the same context.
 links:
   figma: https://www.figma.com/file/noyY6dUMDYjmySpHcMjhkN/HDS-Product---Components?node-id=16384%3A46484&t=XC8SUxxJOFHgqYzK-1
   github: https://github.com/hashicorp/design-system/tree/main/packages/components/addon/components/hds/tabs
+related: ['components/accordian','components/breadcrumb','components/side-nav']
```
Comment (alex-ju) [inline website/docs/components/tabs/index.md:8]:
```suggestion
related: ['components/accordion','components/breadcrumb','components/side-nav']
```

**Review by alex-ju — CHANGES_REQUESTED** [review]
Nice to see these reviewed and aligned with Figma. A few typos make some related links not appear when they should, once those are fixed I'm happy to get it in.

**Review by alex-ju — APPROVED** [review]

**Pull request merged**